src/Chapter3.hs

```diff
@@ -970,6 +1095,45 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Day 
+  = Monday 
+  |  Tuesday 
+  |  Wednesday 
+  |  Thursday 
+  |  Friday 
+  |  Saturday 
+  |  Sunday 
+    deriving (Show, Enum)
+
+
+isWeekend :: Day -> Bool
+isWeekend day = case day of 
+  Saturday -> True 
+  Sunday   -> True 
+  _ -> False 
+
+-- ToDo elegant solution needed
+-- any recommendations 
+nextDay :: Day -> Day
+nextDay day =  case day of
+  Monday    -> Tuesday
+  Tuesday   ->  Wednesday
+  Wednesday   -> Thursday
+  Thursday  -> Friday
+  Friday    -> Saturday
+  Saturday   ->  Sunday
+  Sunday    -> Monday
+
+daysToParty :: Day -> Int 
+daysToParty day = case day of 
+  Monday    -> 4
+  Tuesday   -> 3
+  Wednesday   -> 2
+  Thursday  -> 1
+  Friday    -> 7
+  Saturday  -> 6
+  Sunday    -> 5
```

Comment:
This function also can be implemented a bit shorter by using the `fromEnum` method of the `Enum` typeclass :slightly_smiling_face: